multiplex.go

```diff
@@ -167,6 +167,13 @@ func (mp *Multiplex) NewStream() (*Stream, error) {
 
 func (mp *Multiplex) NewNamedStream(name string) (*Stream, error) {
 	mp.chLock.Lock()
+
+	// We could call IsClosed but this is faster (given that we already have
+	// the lock).
+	if mp.channels == nil {
```

Comment:
@whyrusleeping this is faster (who cares?) but also a bit less reliable (if someone decides to *not* null this out, we'll have problems). Thoughts?